src/realm.js

```diff
@@ -194,6 +194,7 @@ export class Realm {
       output: opts.reactOutput || "create-element",
       hoistableFunctions: new WeakMap(),
       hoistableReactElements: new WeakMap(),
+      immutableObjects: new WeakSet(),
```

Comment:
IMO, we can just add this field directly to ObjectValue but let's check with @hermanventer and @NTillmann if that's ok.